### PR TITLE
Migrate more of the test  to `MergedTree` API

### PR DESCRIPTION
**Pull request by martinvonz**
# Checklist

If applicable:
- [ ] I have updated `CHANGELOG.md`
- [ ] I have updated the documentation (README.md, docs/, demos/)
- [ ] I have updated the config schema (cli/src/config-schema.json)
- [ ] I have added tests to cover my changes

### CODE REVIEW

File: lib/tests/test_commit_builder.rs [hunk 1]
```diff
@@ -14,12 +14,12 @@
```
Comment (yuja) [inline lib/tests/test_commit_builder.rs]: nit: stale subject line in commit description?
> tests: add `create_tree()` returning `MergedTree` and use where easy

File: lib/tests/test_diff_summary.rs [hunk 2]
```diff
@@ -13,7 +13,7 @@
 // limitations under the License.
 
 use jj_lib::matchers::{EverythingMatcher, FilesMatcher};
-use jj_lib::merged_tree::{DiffSummary, MergedTree};
+use jj_lib::merged_tree::DiffSummary;
 use jj_lib::repo_path::RepoPath;
 use test_case::test_case;
 use testutils::TestRepo;
```
Comment (yuja) [inline lib/tests/test_diff_summary.rs:19]: nit: `create_tree()` can be imported by name

**Review by yuja — APPROVED** [review]
Nice

File: lib/tests/test_commit_builder.rs (hunk 1 above)
Comment (martinvonz) [reply lib/tests/test_commit_builder.rs]: Oops, yes. Removed.

File: lib/tests/test_diff_summary.rs (hunk 2 above)
Comment (martinvonz) [reply lib/tests/test_diff_summary.rs]: Done

**Pull request merged**